File: data/classes/queen.py
import pygame
from data.classes.Piece import piece

class queen(piece):
	def __init__(self , pos_piece , color , name):
		super().__init__(pos_piece , color , name)
		img_path = 'data/imgs/' + name + '.png'
		self.img = pygame.transform.scale(pygame.image.load(img_path) , (80 , 80))
		self.notation = 'q'

	def get_possible_move(self , board_config , squares):
		output = []
		for i in range(1 , 8):
			if self.pos[0] - i < 0 or self.pos[1] + i > 7:
				break
			if board_config[self.pos[0] - i][self.pos[1] + i][0] != self.color:
				output.append((self.pos[0] - i , self.pos[1] + i))
			if board_config[self.pos[0] - i][self.pos[1] + i] != '--':
				break

		for i in range(1 , 8):
			if self.pos[0] - i < 0 or self.pos[1] - i < 0:
				break
			if board_config[self.pos[0] - i][self.pos[1] - i][0] != self.color:
				output.append((self.pos[0] - i , self.pos[1] - i))
			if board_config[self.pos[0] - i][self.pos[1] - i] != '--':
				break

		for i in range(1 , 8):
			if self.pos[0] + i > 7 or self.pos[1] - i < 0:
				break
			if board_config[self.pos[0] + i][self.pos[1] - i][0] != self.color:
				output.append((self.pos[0] + i , self.pos[1] - i))
			if board_config[self.pos[0] + i][self.pos[1] - i] != '--':
				break

		for i in range(1 , 8):
			if self.pos[0] + i > 7 or self.pos[1] + i > 7:
				break
			if board_config[self.pos[0] + i][self.pos[1] + i][0] != self.color:
				output.append((self.pos[0] + i , self.pos[1] + i))
			if board_config[self.pos[0] + i][self.pos[1] + i] != '--':
				break

		for i in range(1 , 8):
			if self.pos[0] + i > 7:
				break
			if board_config[self.pos[0] + i][self.pos[1]][0] != self.color:
				output.append((self.pos[0] + i , self.pos[1]))
			if board_config[self.pos[0] + i][self.pos[1]] != '--':
				break

		for i in range(1 , 8):
			if self.pos[0] - i < 0:
				break
			if board_config[self.pos[0] - i][self.pos[1]][0] != self.color:
				output.append((self.pos[0] - i , self.pos[1]))
			if board_config[self.pos[0] - i][self.pos[1]] != '--':
				break

		for i in range(1 , 8):
			if self.pos[1] + i > 7:
				break
			if board_config[self.pos[0]][self.pos[1] + i][0] != self.color:
				output.append((self.pos[0] , self.pos[1] + i))
			if board_config[self.pos[0]][self.pos[1] + i] != '--':
				break


		for i in range(1 , 8):
			if self.pos[1] - i < 0:
				break	
			if board_config[self.pos[0]][self.pos[1] - i][0] != self.color:
				output.append((self.pos[0] , self.pos[1] - i))
			if board_config[self.pos[0]][self.pos[1] - i] != '--':
				break

		Oput = []
		for pos in output:
			for square in squares:
				if square.pos == pos:
					Oput.append(square)
					break

		return Oput

File: data/object/Queen.cpp
#include "Queen.h"


Queen::Queen(std::string notation, sf::Vector2i coordinate, float size, sf::Texture* texture)
	: Piece(notation, coordinate, size, texture)
{

}

std::vector<sf::Vector2i> Queen::getPossibleMove(std::string config[][8] , Engine *game)
{
	std::vector<sf::Vector2i> output;
	int x = this->coordinate.x;
	int y = this->coordinate.y;
	for (int i = x - 1; i >= 0; i--)
	{
		if (config[y][i] == "--")
		{
			output.push_back(sf::Vector2i(i, y));
		}
		else if (config[y][i][0] == this->color)
			break;
		else if (config[y][i][0] != this->color)
		{
			output.push_back(sf::Vector2i(i, y));
			break;
		}
	}
	for (int i = x + 1; i < 8; i++)
	{
		if (config[y][i] == "--")
		{
			output.push_back(sf::Vector2i(i, y));
		}
		else if (config[y][i][0] == this->color)
			break;
		else if (config[y][i][0] != this->color)
		{
			output.push_back(sf::Vector2i(i, y));
			break;
		}
	}

	for (int i = y - 1; i >= 0; i--)
	{
		if (config[i][x] == "--")
		{
			output.push_back(sf::Vector2i(x, i));
		}
		else if (config[i][x][0] == this->color)
			break;
		else if (config[i][x][0] != this->color)
		{
			output.push_back(sf::Vector2i(x, i));
			break;
		}
	}

	for (int i = y + 1; i < 8; i++)
	{
		if (config[i][x] == "--")
		{
			output.push_back(sf::Vector2i(x, i));
		}
		else if (config[i][x][0] == this->color)
			break;
		else if (config[i][x][0] != this->color)
		{
			output.push_back(sf::Vector2i(x, i));
			break;
		}
	}
	for (int i = 1; i < 8; i++)
	{
		int x = this->coordinate.x + i;
		int y = this->coordinate.y + i;
		if (x < 0 || x >= 8 || y < 0 || y >= 8)
			break;

		if (config[y][x] == "--")
		{
			output.push_back(sf::Vector2i(x, y));
		}
		else if (config[y][x][0] == this->color)
			break;
		else if (config[y][x][0] != this->color)
		{
			output.push_back(sf::Vector2i(x, y));
			break;
		}
	}

	for (int i = 1; i < 8; i++)
	{
		int x = this->coordinate.x - i;
		int y = this->coordinate.y - i;
		if (x < 0 || x >= 8 || y < 0 || y >= 8)
			break;

		if (config[y][x] == "--")
		{
			output.push_back(sf::Vector2i(x, y));
		}
		else if (config[y][x][0] == this->color)
			break;
		else if (config[y][x][0] != this->color)
		{
			output.push_back(sf::Vector2i(x, y));
			break;
		}
	}

	for (int i = 1; i < 8; i++)
	{
		int x = this->coordinate.x - i;
		int y = this->coordinate.y + i;
		if (x < 0 || x >= 8 || y < 0 || y >= 8)
			break;

		if (config[y][x] == "--")
		{
			output.push_back(sf::Vector2i(x, y));
		}
		else if (config[y][x][0] == this->color)
			break;
		else if (config[y][x][0] != this->color)
		{
			output.push_back(sf::Vector2i(x, y));
			break;
		}
	}

	for (int i = 1; i < 8; i++)
	{
		int x = this->coordinate.x + i;
		int y = this->coordinate.y - i;
		if (x < 0 || x >= 8 || y < 0 || y >= 8)
			break;

		if (config[y][x] == "--")
		{
			output.push_back(sf::Vector2i(x, y));
		}
		else if (config[y][x][0] == this->color)
			break;
		else if (config[y][x][0] != this->color)
		{
			output.push_back(sf::Vector2i(x, y));
			break;
		}
	}

	return output;
}